src/ucp/rndv/rndv_rtr.c

```diff
@@ -53,7 +53,8 @@ ucp_proto_rndv_rtr_common_init(const ucp_proto_init_params_t *init_params,
         .super.hdr_size      = sizeof(ucp_rndv_rtr_hdr_t),
         .super.send_op       = UCT_EP_OP_AM_BCOPY,
         .super.memtype_op    = UCT_EP_OP_LAST,
-        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_RESPONSE,
+        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_RESPONSE |
+                               UCP_PROTO_COMMON_INIT_FLAG_ADJ_MAX_FRAG,
```

Comment:
looking at this, i think it can limit the size of RTR.size field and this is not what we want.
if it really happens, need to keep using a flag (and not send_op) and just remove this flag from here.

Reply:
did not get this comment, can you please elaborate?

Reply:
rtr protocol is about fetching data from remote. so need to make sure this will not cause the size of the fetched data to be reduced.

Reply:
It means that we cannot use `send_op` for the purpose. `ucp_rndv_rtr_proto` has `send_op` equals to `UCT_EP_OP_AM_BCOPY`. But we do not want to limit RTR size. Than the flag can be used. And it should be removed from flags of protocol  `ucp_rndv_rtr_proto`.

Reply:
Removed.

Reply:
"It means that we cannot use send_op for the purpose. "
can you pls check it's really the case (i am not sure). Also can you pls check it for rndv/rts protocol as well and remove from there if needed.

Reply:
but we are sending rts and rtr as AM, why is seg_size limitation not relevant?

Reply:
> but we are sending rts and rtr as AM, why is seg_size limitation not relevant?

Currently, fragment size affects protocol selection thresholds. In this PR we take into account segment size while retrieving fragment size. We choose protocol depending on message size. However, we don't take into account fragment size when we send RTS/RTR. Therefore, segment size is not counted.
The only place was in https://github.com/openucx/ucx/blob/db3606d6826c6c69692df22fdbfaa9dd23a1f4f1/src/ucp/core/ucp_am.c#L763

At least we can add an assert. E.g. here https://github.com/openucx/ucx/blob/db3606d6826c6c69692df22fdbfaa9dd23a1f4f1/src/ucp/proto/proto_single.inl#L41

Reply:
let's not handle ctrl message size limit for now, we'll do it when adding rndv/iov support

Reply:
ok, 
@brminich, @yosefe do you have any other comments to this PR?